.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: ci
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        # currently windows would be difficult to support due to the build
+        # requiring autotools
+        os: [macos, ubuntu]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: "setup linux build environment"
+      if: ${{ matrix.os == 'ubuntu' }}
+      run: sudo apt install -y protobuf-compiler libprotobuf-dev libutempter-dev
```

Comment:
Is it really the case you don't need autotools here? https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-Readme.md doesn't seem to list it. I would expect you to need to explicitly install autoconf and automake.

Reply:
I'm not sure why the actions [ran fine][1] without these, but I went ahead and added them just to be safe - similarly to you, I had some reservations about not explicitly installing them. I was a bit confused after reading [this][2], since I thought that meant that autotools stuff was installed by default.

[1]: https://github.com/ezzieyguywuf/mosh/actions/runs/1767435406
[2]: https://github.com/actions/starter-workflows/issues/53